ford/sourceform.py

```diff
@@ -775,7 +823,15 @@ def __init__(
             elif self.BLOCK_RE.match(line):
                 blocklevel += 1
             elif self.ASSOCIATE_RE.match(line):
-                associatelevel += 1
+                # Associations 'call' the rhs of the => operator
+                self._add_procedure_calls(line, associations)
+
+                # Register the associations
+                assoc_str = ford.utils.strip_paren(
+                    self.ASSOCIATE_RE.match(line).group(2), 0
```

Comment:
You could make this even better by using `match := self.ASSOCIATE_RE.match(line)` in the `elif` so we can reuse `match`, and then also naming the group in `ASSOCIATE_RE` so we can do `match[<group name>]` here